CMakeLists.txt

```diff
@@ -143,6 +144,10 @@ if (EKAT_ENABLE_TESTS)
   add_subdirectory(tests)
 endif()
 
+if (EKAT_ENABLE_CUDA_MEMCHECK AND NOT Kokkos_ENABLE_CUDA)
```

Comment:
Should we also check that one does not request both valgrind and cuda-memcheck at the same time? The macro `EkatCreateUnitTest` would pick valgrind in that case, while that's probably not what the user wants.

Reply:
Good call